joomla/joomla-cms
# UX Improvement - client-side form validation

UX Improvement - client-side form validation + com_users frontend label strings

<h2>How to Test ?</h2>

Note: test example with registration form. You can test with another form (profil, contact, third party form using Joomla client-side form validate...)

<h3>Before Patch</h3>

<h4>Test</h4>
- Resize you window to be around 800px height.
- Use login form, and click on "Create an account"
- Click directly on "Register" (by default on a fresh install you have 6 fields required, so you will have 6 fields invalid)

<h4>Result</h4>
- The page don't go up to the system message container.
- Ordering of fields invalid is wrong
- You have ":" at the end of the labels

<h3>After Patch</h3>

<h4>More than 5 fields required not filled</h4>
- Auto-scroll to the system message container
- No Field invalid + label displayed (in case of a form with 20 fields required, not filled, could be better than a long list! ;-) )
- Single content : "All fields with an \* are required."

<h4>Less than 5 fields required not filled</h4>
- Auto-scroll to the system message container
- Ordering is the same as the form fields order
- you don't have the ":" at the end of label

<h4>In all cases</h4>
- System message container header : "Please make sure the form is complete and valid."

UX Improvement :
- correct order of invalid field
- auto-scroll to system message container if form not valid
- if more than 5 fields are invalid, do not list all invalid field. Replaced by a single alert message.
- 2 strings added for client-side form validation (title + single message if more than 5 fields invalid)

Code Standards & Consistency :
- clean all indentation (previously mix of 1 space + 1 tab per indentation)
- a few changes to follow Joomla Javascript Code Standards (http://joomla.github.io/coding-standards/?coding-standards/chapters/javascript.md)
- remove all ":" at the end of labels, for consistency with all form fields, and to improve rendering of client-side form validation (no ":" for label in the Field invalid : %s)
